theano/sandbox/cuda/dnn.py

```diff
@@ -256,7 +256,8 @@ def make_node(self, img_shape, kern_shape):
             raise TypeError('kern must be 1D shape tensor')
 
         return Apply(self, [img_shape, kern_shape],
-                     [CDataType("cudnnConvolutionDescriptor_t")()])
+                     [CDataType("cudnnConvolutionDescriptor_t",
+                                freefunc="cudnnDestroyConvolutionDescriptor")()])
```

Comment:
this do not trigger a new compilation, so to have the fix, we need to delete the compiledir.

The easy fix, to bump those op c_code_cache_version. The better fix, to add in CDataType.c_code_cache_version, the cdata type and freefunc.

I made a PR to your branch: https://github.com/abergeron/Theano/pull/20